Entity/Components/HealthComponent.gd

```diff
@@ -7,32 +7,30 @@ class_name HealthComponent
 ## Emit when health is changed
 signal on_health_changed(new_health: int)
 
+signal on_block_changed(new_block: int)
+
 ## The maximum health the entity can have; this is a hard limit
-@export var max_health: float = 100
+@export var max_health: int = 100
 ## The health the entity currently has [br]
 ## Between 0 and max_health
-var current_health: float = 100 # ? change this to max health, or just unset since it's init by the ready
+var current_health: int = 100 # ? change this to max health, or just unset since it's init by the ready
+
+
+var current_block: int = 0
 
 ## Initialize the health component, putting health to max health
 ## This only happens at game start for the player, the value is then tracked inside the [PlayerManager] singleton [br]
 ## For enemies, this is called when the scene is instantiated (or more accurately, when each enemy is instanciated) [br]
 func _ready() -> void:
 	_set_health(max_health)
+	PhaseManager.on_phase_changed.connect(reset_block_on_round_start)
```

Comment:
How does GoDot decide what order to apply signals in ? 

For example, we have PhaseManager.on_phase_changed connected to this, but also to player start turn in battler. 
If a player has a buff that would increase his block on round start for example, would it be possible it happens before this reset and so it will then be wrongly reset to 0 ?

Reply:
that's a good question, I'm not sure how Godot decides, and it indeed might be an issue. To solve the issue we could instead decide to reset the block on enemy turn finishing instead of player turn start.

Another question we could wonder is with effects that apply at turn start like poison. Is poison blocked by shield ? If yes, does poison trigger before or after the block reset, in which case is shield from the previous turn could block poison or not. Moving this to enemy turn end doesn't really solve the problem, because an enemy could die from poison thus finishing the fight before the player takes poison damage.

Need to check how signals are done, but my intuition tells me it's just "whatever component gets picked first", in which case we might actually need to create a new phase to handle those cases

Reply:
I was trying to find what ordering there is but wasn't really able to. 

I think most likely is that rather then it being random, what matters is the component's position in the tree. If that was the case (which I am not sure of) that would at least take the randomness away but it still doesn't resolve the problem. 

We would need some way to more clearly separate the specific cases. The only issue is I am not sure how many different cases that need to be resolved separated there are at the moment, and there might be more in the future ?

Reply:
If we only need new signals, then I could just make a simple `before_phase_change` signal in the `PhaseManager`. It might be too simple, but anything more than that would probably be too complex for this PR, and we can do something more complicated if needed later

Reply:
I think adding a phase shouldn't be too difficult (?) and would solve the problem

I'll open an issue to change the way we change phases so it's clearer later, but for now you will need to find where we call the change from phase enemy finishing to player attacking and insert a "remove block" phase between those two.

What do you think of this solution ?

Reply:
Resolving this for now as there is temp solution + actual solution planned for future PR (https://github.com/Saplings-Projects/1M_sub/issues/120)